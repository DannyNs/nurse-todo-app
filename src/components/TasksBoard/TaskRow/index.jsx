import React from 'react';
import PropType from 'prop-types';
import { observer } from 'mobx-react';

import './index.scss';

const TaskRow = observer(({
  id, name, description, time, done, markAsDone,
}) => (
  <div className="nr-task-row">
    <div className="nr-task-row__labels">
      <div>Name:</div>
      <div>Description:</div>
      <div>Date:</div>
      <div>Done:</div>
      <div>Button:</div>
    </div>
    <div className="nr-task-row__content">
      <div>
        <span>{id}</span>
        {name}
      </div>
      <div>{description}</div>
      <div>{new Date(time * 1000).toUTCString()}</div>
      <div>{done ? 'yes' : 'no'}</div>
      <div>
        <button type="button" onClick={markAsDone}>
          Done
        </button>
      </div>
    </div>
  </div>
));

TaskRow.propTypes = {
  id: PropType.number.isRequired,
  name: PropType.string.isRequired,
  description: PropType.string.isRequired,
  time: PropType.number.isRequired,
  markAsDone: PropType.func.isRequired,
};

export default TaskRow;
