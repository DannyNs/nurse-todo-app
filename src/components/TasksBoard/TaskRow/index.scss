.nr-task-row {
  
  &__done {
    color: var(--main-success-color);
  }

  &__name {
    color: #000000;
    font-weight: bold;
  }
}