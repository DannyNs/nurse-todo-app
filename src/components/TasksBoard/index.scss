.hn-tasks-board {

  &__child {
    margin: 20px;
  }
}