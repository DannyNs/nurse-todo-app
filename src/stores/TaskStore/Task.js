import {
  observable, action,
} from 'mobx';

export default class Task {
  id = null;

  name = '';

  description = '';

  time = new Date().getTime();

  @observable
  done = false;

  constructor(source) {
    Object.assign(this, source);
  }

  @action markAsDone = () => {
    this.done = true;
  }
}
