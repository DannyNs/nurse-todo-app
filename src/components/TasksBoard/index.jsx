import React from 'react';
import PropType from 'prop-types';
import { observer } from 'mobx-react';
import Task from '../../stores/TaskStore/Task';

import TaskRow from './TaskRow';

import './index.scss';

const TasksBoard = observer(({ tasks }) => (
  <div className="hn-Tasks-board">
    {
      tasks
        .map(({
          id, name, description, time, done, markAsDone,
        }) => (
          <div className="hn-Tasks-board__child" key={id}>
            <TaskRow
              id={id}
              name={name}
              description={description}
              time={time}
              done={done}
              markAsDone={markAsDone}
            />
          </div>
        ))
    }
  </div>
));

TasksBoard.propTypes = {
  tasks: PropType.arrayOf(PropType.instanceOf(Task)).isRequired,
};

export default TasksBoard;
