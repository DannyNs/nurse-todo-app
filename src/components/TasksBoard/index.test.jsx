import React from 'react';
import { shallow } from 'enzyme';
import TasksBoard from '.';
import Task from '../../stores/TaskStore/Task';

describe('TasksBoard', () => {
  it('should render without crashing', () => {
    // give
    const tasks = [
      new Task({
        id: 1,
        name: 'Brian Adams',
        description: 'Change pillow',
        time: new Date().getTime(),
        done: true,
      }),
      new Task({
        id: 2,
        name: 'Bob Dylan',
        description: 'Make bed',
        time: new Date().getTime() + 1000,
        done: false,
      }),
    ];

    // when
    const wrapper = shallow(
      <TasksBoard tasks={tasks} />,
    );

    // then
    expect(wrapper).toMatchSnapshot();
  });
});
