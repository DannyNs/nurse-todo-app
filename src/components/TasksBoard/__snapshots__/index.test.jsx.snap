// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksBoard should render without crashing 1`] = `
<div
  className="hn-Tasks-board"
>
  <div
    className="hn-Tasks-board__child"
    key="1"
  >
    <_class
      description="Change pillow"
      done={true}
      id={1}
      markAsDone={[Function]}
      name="Brian Adams"
      time={1542907030618}
    />
  </div>
  <div
    className="hn-Tasks-board__child"
    key="2"
  >
    <_class
      description="Make bed"
      done={false}
      id={2}
      markAsDone={[Function]}
      name="Bob Dylan"
      time={1542907031622}
    />
  </div>
</div>
`;
